From a9dd85297e4deaf76cad2c598dc9d496c81e88a5 Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Tue, 23 Oct 2012 16:24:30 +0200
Subject: [PORT FROM MAIN] Allow AudioHAL Generic and stub to compile

BZ: 66390

Virtual fonction on StreamIn were introduced without implementation
in the AudioHAL stub and generic.

This patch keeps the ability to compile of these 2 components.

Feature: AudioComms
Category: Integration
Purpose: ABSP-Adapter
Origin: Internal
Upstream-candidate: yes
Change-Id: Iaead43e1ca609f7b844af5e2dea723dbf20679a6
Orig-Change-Id: Ie186a7ef1ac573d7f256bbfa5f55ce6a01d3c8a9
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 audio/AudioHardwareGeneric.h |    3 ++-
 audio/AudioHardwareStub.h    |    2 ++
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/audio/AudioHardwareGeneric.h b/audio/AudioHardwareGeneric.h
index 7b41e95..acdbc93 100644
--- a/audio/AudioHardwareGeneric.h
+++ b/audio/AudioHardwareGeneric.h
@@ -92,7 +92,8 @@ public:
     virtual status_t    setParameters(const String8& keyValuePairs);
     virtual String8     getParameters(const String8& keys);
     virtual unsigned int  getInputFramesLost() const { return 0; }
-
+    virtual status_t addAudioEffect(effect_handle_t effect) { return NO_ERROR; }
+    virtual status_t removeAudioEffect(effect_handle_t effect) { return NO_ERROR; }
 private:
     AudioHardwareGeneric *mAudioHardware;
     Mutex   mLock;
diff --git a/audio/AudioHardwareStub.h b/audio/AudioHardwareStub.h
index 0858f37..c5f7a80 100644
--- a/audio/AudioHardwareStub.h
+++ b/audio/AudioHardwareStub.h
@@ -58,6 +58,8 @@ public:
     virtual status_t    setParameters(const String8& keyValuePairs) { return NO_ERROR;}
     virtual String8     getParameters(const String8& keys);
     virtual unsigned int  getInputFramesLost() const { return 0; }
+    virtual status_t addAudioEffect(effect_handle_t effect) { return NO_ERROR; }
+    virtual status_t removeAudioEffect(effect_handle_t effect) { return NO_ERROR; }
 };
 
 class AudioHardwareStub : public  AudioHardwareBase
-- 
1.7.4.1

